src/middlewared/middlewared/etc_files/web_ui_root_login_alert.py

```diff
@@ -0,0 +1,7 @@
+
+
+async def render(service, middleware, render_ctx):
```

Comment:
Agree with Andrew. `render_ctx` should already have user/group query results so it'd be best to pass that through to the other functions to keep from having to unnecessary datastore.query calls.